R/image.R

```diff
@@ -124,8 +127,18 @@ pixdim.default <- function (object)
         return (attr(object, "pixdim"))
     else if (!is.null(dim(object)))
         return (rep(1, length(dim(object))))
-    else
+    else {
+        pdim = tryCatch({
+            hdr = niftiHeader(object)
+            d1 = abs(hdr$dim[1])
+            hdr$pixdim[seq(2, 2 + d1 - 1)]
+        })
+        if (!inherits(pdim, "try-error")) {
+            return(pdim)
+        }
```

Comment:
Thanks for catching this – the same basic issue also applies to `pixunits`. I think the logic should be a little different, and pushed down into the C++ code for consistency, so I'll have another stab at this and mention issue #12 so that you see it.